### ISSUE TITLE
Re-inject the chaos, when the pod restart

### ISSUE THREAD

**Issue by xlgao-zju** [chaos-mesh/chaos-mesh#3867]
## Feature Request

**Is your feature request related to a problem? Please describe:**

For now, some kinds of chaos will recover after the pod restart, e.g. IO chaos.

**Describe the feature you'd like:**

Re-inject the chaos, when the pod restart.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by xlgao-zju** [comment]
cc @STRRL

**Comment by STRRL** [comment]
also ref: https://github.com/chaos-mesh/chaos-mesh/issues/1259

**Comment by xlgao-zju** [comment]
After discussed with @STRRL offline. I think maybe we can create a controller. 

In this controller, we watch pod and all chaos, and we can build a map to store the relation of  chaos and pod(selected by chaos). And when `restartCount` of pod get increased, we can make the related chaos re-inject.

As for the trigger of chaos re-injection,  we can add a annotation (maybe called need-re-injection) on chaos.

**Comment by g1eny0ung** [comment]
It seems like a useful feature, but I think we need to look at it more thoroughly, for example, is there a conflict between `PodChaos` and this feature?

Also, if this feature can be pushed forward, I think it's better to draft an RFC to discuss more details. Because it is involved in almost all types of chaos.

**Comment by xlgao-zju** [comment]
@g1eny0ung yep, support all types of chaos is a big task and need lots of test. Maybe we can start from stress chaos or IO chaos.

**Comment by g1eny0ung** [comment]
> @g1eny0ung yep, support all types of chaos is a big task and need lots of test. Maybe we can start from stress chaos or IO chaos.

Agreed. Starting with `IOChaos` or `StressChaos` is a good place to start.

**Comment by STRRL** [comment]
> In this controller, we watch pod and all chaos, and we can build a map to store the relation of chaos and pod(selected by chaos). And when `restartCount` of pod get increased, we can make the related chaos re-inject.

Is there possible to put that "map" as a part of Kind Status? Comparing with a map inside of chaos-controller-manager's memory, it seems status is a more transparent place to inspect. And it might be also a little bit easier for disaster recovery with for HA chaos mesh controller

**Comment by xlgao-zju** [comment]
> Is there possible to put that "map" as a part of Kind Status?

Yes, it will be better. But is there any good place to store the "map". And lock is needed for the "map".
